lib/AST/ASTDumper.cpp

```diff
@@ -708,13 +708,9 @@ namespace {
       printCommon(PD, "protocol");
 
       OS << " requirement signature=";
-      if (PD->isRequirementSignatureComputed()) {
-        OS << GenericSignature::get({PD->getProtocolSelfType()} ,
-                                    PD->getRequirementSignature())
-                ->getAsString();
-      } else {
-        OS << "<null>";
-      }
+      OS << GenericSignature::get({PD->getProtocolSelfType()} ,
+                                   PD->getRequirementSignature())
```

Comment:
This is the one place I can think of where we still want to check `isRequirementSignatureComputed()`, because we want the AST dumper to dump only what's currently been realized in the AST... and avoid performing extra computation (in this case, a bunch of type checking) that makes it harder to debug.

Reply:
Alright, I brought back the original version.